core/src/main/java/org/elasticsearch/discovery/local/LocalDiscovery.java

```diff
@@ -308,7 +273,7 @@ public void removeListener(InitialStateDiscoveryListener listener) {
 
     @Override
     public String nodeDescription() {
-        return clusterName.value() + "/" + localNode.id();
+        return clusterName.value() + "/" + localNode();
```

Comment:
This will append `DiscoveryNode#toString` but I still think that you want `DiscoveryNode#id`?